### PR TITLE
fix(SearchableEncryption): disable shared cached

### PR DESCRIPTION
**Pull request by texastony**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by ajewellamz — APPROVED** [review]

File: DynamoDbEncryption/dafny/DynamoDbEncryption/src/ConfigToInfo.dfy [hunk 1]
```diff
@@ -119,7 +119,21 @@ module SearchConfigToInfo {
...
+      && output.Failure?
+         // If the failure was NOT caused by booting up the MPL
+      && !output.error.AwsCryptographyMaterialProviders?
+      ==>
+        && output.error.DynamoDbEncryptionException?
+        && output.error.message == "Searchable Encryption does not support the Shared Cache type at this time."
   {
+    // TODO-FutureCleanUp : It is not-good that the MPL is initialized here;
+    // The MPL has a config object that could hold customer intent that affects behavior.
+    // Today, it does not. But tomorrow?
```
Comment (texastony) [inline DynamoDbEncryption/dafny/DynamoDbEncryption/src/ConfigToInfo.dfy:136]: I need to create a GHI for this.

File: DynamoDbEncryption/dafny/DynamoDbEncryption/src/ConfigToInfo.dfy [hunk 2]
```diff
@@ -119,7 +119,21 @@ module SearchConfigToInfo {
...
+    ensures
+      && config.multi?
+      && config.multi.cache.Some?
+      && config.multi.cache.value.Shared?
+      && output.Failure?
+         // If the failure was NOT caused by booting up the MPL
+      && !output.error.AwsCryptographyMaterialProviders?
+      ==>
+        && output.error.DynamoDbEncryptionException?
+        && output.error.message == "Searchable Encryption does not support the Shared Cache type at this time."
```
Comment (RitvikKapila) [inline DynamoDbEncryption/dafny/DynamoDbEncryption/src/ConfigToInfo.dfy:132]:
```suggestion
   ensures
      && config.multi?
      && config.multi.cache.Some?
      && config.multi.cache.value.Shared?
        ==> 
        && output.Failure?
        // If the failure was NOT caused by booting up the MPL
        && !output.error.AwsCryptographyMaterialProviders?
        ==>
          && output.error.DynamoDbEncryptionException?
          && output.error.message == "Searchable Encryption does not support the Shared Cache type at this time."

**Review by RitvikKapila — APPROVED** [review]
LGTM

**Comment by texastony** [comment]
I have never seen this test fail:
```
software.amazon.cryptography.examples.migration.awsdbe.encrypttable.TestEncryptExistingTable > TestEncryptExistingTable FAILED
    java.lang.AssertionError
        at software.amazon.cryptography.examples.migration.awsdbe.encrypttable.TestEncryptExistingTable.EncryptExistingTable(TestEncryptExistingTable.java:179)
        at software.amazon.cryptography.examples.migration.awsdbe.encrypttable.TestEncryptExistingTable.TestEncryptExistingTable(TestEncryptExistingTable.java:197)
```

I wonder what is up...

**Pull request merged**